docs/api/en/helpers/BoxHelper.html

```diff
@@ -28,10 +28,25 @@ <h2>Example</h2>
 
 
 		<code>
-		var sphere = new THREE.SphereGeometry();
-		var object = new THREE.Mesh( sphere, new THREE.MeshBasicMaterial( 0xff0000 ) );
-		var box = new THREE.BoxHelper( object, 0xffff00 );
-		scene.add( box );
+		// Creating the object whose bounding box we want to compute
+		var sphereGeom = new THREE.SphereGeometry();
+		var sphereObject = new THREE.Mesh( sphereGeom, new THREE.MeshBasicMaterial( 0xff0000 ) );
+
+		// Creating the helper and displaying it in the scene
+		var helper = new THREE.BoxHelper( sphereObject, 0xffff00 );
+		scene.add( helper );
+
+		// Creating the actual bounding box (on which we can test intersection with other Box3 objects)
+		var box3 = new THREE.Box3();
+		// Shape and position the box after the object
+		box3.setFromObject( sphere );
+
+		// Let's move the sphere
+		sphereObject.position.set( 7, 7, 7 );
+
+		// Whenever the object is moved or rotated or scaled, we update both the bounding box and the helper
+		helper.update(); 
+		box3.setFromObject( sphereObject );
```

Comment:
Usually you do not want generate a new bounding box each time the object is updated. Instead, you would copy the bounding box from the geometry and just apply the world matrix of the object.
```js
sphereGeom.computeBoundingBox(); // just do this once

// in your animation loop

box.copy( object.geometry.boundingBox ).applyMatrix4( object.worldMatrix );
```
You would do the same with bounding sphere.

Reply:
Didn't know `setFromObject` had the effect of creating a new bounding box from scratch instead of copying the relevant parameters only. Thanks for the insight. I'm gonna change the example as you suggest, but may I ask why don't we change `setFromObject` to do those operations instead of re-creating a new Box3 object every time? Are there cases when that is the intended behavior?

Also, is the World matrix what is commonly referred to as a Model matrix (mapping from model coordinates to world coordinates, containing all the transformations)?

Reply:
>  but may I ask why don't we change setFromObject to do those operations instead of re-creating a new Box3 object every time?

`Box3.setFromObject()` should compute an AABB from scratch by evaluating geometry data. I referred in my previous post to the case where you apply a transformation matrix to an _existing_ AABB. These are two different operations and both are needed in different scenarios.

> Also, is the World matrix what is commonly referred to as a Model matrix

Yep.

Reply:
Understood, thanks for the clarification. I'm gonna do all of the requested changes.